examples/images_contours_and_fields/pcolormesh_grids.py

```diff
@@ -0,0 +1,141 @@
+"""
+============================
+Pcolormesh grids and shading
+============================
+
+`.axes.Axes.pcolormesh` and `~.axes.Axes.pcolor` have a few options for
+how grids are laid out and the shading between the grid points.
+
+Generally, if *Z* is size *MxN* then the grid *X* and *Y* can be
+specified with either size *M+1 x N+1* or *MxN*, depending on the
+argument for the ``shading`` keyword argument.
+
+"""
+
+import matplotlib
+import matplotlib.pyplot as plt
+from matplotlib.colors import BoundaryNorm
+from matplotlib.ticker import MaxNLocator
+import numpy as np
+
+###############################################################################
+# Flat Shading
+# ------------
+#
+# The grid specification with the least assumptions is ``shading='flat'``
+# and if the grid is one larger than the data in each dimesion, i.e. has size
+# *M+1 x N+1*.  In that case *X* and *Y* sepcify the corners of quadrilaterals
+# that are colored with the values in *Z*:
+#
+#        (X[i+1, j], Y[i+1, j])       (X[i+1, j+1], Y[i+1, j+1])
+#                          +-----+
+#                          |     |
+#                          +-----+
+#        (X[i, j], Y[i, j])       (X[i, j+1], Y[i, j+1])
+#
+# Here we specify the edges of the *3x5* quadirlaterals with *X* and *Y* that
+# are *4x6*.
+np.random.seed(19680801)
+Z = np.random.rand(3, 5)
+x = np.arange(6)  # len = 6
+y = np.arange(4)  # len = 4
+
+fig, ax = plt.subplots()
+ax.pcolormesh(x, y, Z, shading='flat', vmin=Z.min(), vmax=Z.max())
+def _annote(ax, x, y, title):
+    # this all gets repeated below:
+    X, Y = np.meshgrid(x, y)
+    ax.plot(X.flat, Y.flat, 'o', color='m')
+    ax.set_xlim(-0.7, 5.2)
+    ax.set_ylim(-0.7, 3.2)
+    ax.set_title(title)
+_annote(ax, x, y, "shading='flat'")
+
+
+###############################################################################
+# Flat Shading, same size grid
+# ----------------------------
+#
+# Often, however, data is provided where *X* and *Y* are the same size as *Z*.
```

Comment:
Z is 3x5, right?

Reply:
Yes, but X and Y are **not**, they are 1-dimensional, which was what I was getting at here.

Reply:
Will `match the size of *Z*` work to make it clear the sizes correspond?

Reply:
Maybe put correspond into the text? `... match the corresponding size of *Z*`? This may be making things more confusing though... I may be sacrificing **correct** for readability here and not improving things. I'm fine with your `match` addition rather than `same`.

Reply:
I've added this to the top of the example: 
```
Note that below we specify vectors *x* as either length N or N+1 and *y* as length 
M or M+1, and `~.axes.Axes.pcolormesh` makes into the mesh matrices *X* and *Y.  
```

Hopefully that helps clarify the conversational shorthands below?

Reply:
Yes, that is a nice addition!

Suggested rewording: `... pcolormesh internally makes the mesh matrices *X* and *Y* from the input vectors.`